homeassistant/components/binary_sensor/isy994.py

```diff
@@ -49,22 +70,183 @@ def setup_platform(hass, config: ConfigType,
 
 
 class ISYBinarySensorDevice(isy.ISYDevice, BinarySensorDevice):
-    """Representation of an ISY994 binary sensor device."""
+    """Representation of an ISY994 binary sensor device.
+
+    Often times, a single device is represented by multiple nodes in the ISY,
+    allowing for different nuances in how those devices report their on and
+    off events. This class turns those multiple nodes in to a single Hass
+    entity and handles both ways that ISY binary sensors can work.
+    """
 
     def __init__(self, node) -> None:
         """Initialize the ISY994 binary sensor device."""
-        isy.ISYDevice.__init__(self, node)
+        super().__init__(node)
+        self._negative_node = None
+        self._heartbeat_node = None
+        self._heartbeat_timestamp = None
+        self._device_class_from_type = self._detect_device_type()
+        # pylint: disable=protected-access
+        if self._node.status._val == -1:
+            self._computed_state = None
+        else:
+            self._computed_state = bool(self._node.status._val)
+
+    @asyncio.coroutine
+    def async_added_to_hass(self) -> None:
+        """Subscribe to the node and subnode event emitters."""
+        super().async_added_to_hass()
```

Comment:
This is a coroutine, so you should `yield from` it to schedule it.

Reply:
Thanks for the help here - I am a little unclear on when to yield in a coroutine. Are you saying that I should `yield from` the super() call, but the other calls in this method are fine?

Reply:
Yes!